portfolio/src/main/java/com/google/sps/servlets/Event.java

```diff
@@ -1,31 +1,63 @@
-package com.google.sps;
+package com.google.sps.servlets;
 import java.util.*;
 
 public final class Event{
     private final String organizer;
     private final String eventName;
     private final String location;
     private final String description;
-    private final Date date;
+    private final Date start;
+    private final Date end;
+    private final String password;
 
-    public Event(String organizer, String eventName, String location, String description, Date date){
+    public Event(String organizer, String eventName, String location, String description, Date start, Date end){
         this.organizer = organizer;
         this.eventName = eventName;
         this.location = location;
         this.description = description;
-        this.date = date;
+        this.start = start;
+        this.end = end;
+        this.password = generatePassword();
```

Comment:
I'm a little confused about how this works. If a new password is generated every time we turn the Entity into an Event, how do the passwords ever match? I may be missing something, but it seems like this is the sequence of events that would happen:

1. I create an event and get password "abc"
2. "abc" gets stored in datastore as part of the event
3. I load the page for this event
4. When the event gets loaded and gets converted from Entity to Event, a new password "def" is generated in this constructor
5. I enter "abc" but it says that's wrong because it thinks the password is "def"

Reply:
Yep, I realized this was a problem and made an update in a later commit ([https://github.com/jamilao/sps-team-project/pull/10/commits/b83c4fbbcdd9025e65ec1100740398f5b924042c]) that stores generated passwords when posting new events. These stored passwords are then explicitly set when getting events from datastore, which would look something like:

4. When the event gets loaded and gets converted from Entity to Event, a new password "def" is generated in this constructor.
5. The password "abc" stored in datastore is set on the loaded Event, replacing "def".
6. I enter "abc" which would be recognized as correct.